packages/ai-mcp/src/browser/mcp-frontend-application-contribution.ts

```diff
@@ -83,6 +87,18 @@ export class McpFrontendApplicationContribution implements FrontendApplicationCo
         });
         this.frontendMCPService.registerToolsForAllStartedServers();
     }
+    protected async autoStartServers(servers: Map<string, MCPServerDescription>): Promise<void> {
+        const serverNames = servers.keys();
+        const startedServers = await this.frontendMCPService.getStartedServers();
+        for (const name of serverNames) {
+            const serverDesc = servers.get(name);
+            if (serverDesc && serverDesc.autostart) {
+                if (!startedServers.includes(name)) {
+                    await this.frontendMCPService.startServer(name);
+                }
+            }
+        }
```

Comment:
Minor: Seems weird to specifically iterate over all keys but to retrieve all their associated values anyway. We could just iterate over all entries in one go.